catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/SubscriptionsPersistentStoreImpl.java

```diff
@@ -0,0 +1,220 @@
+package org.codice.ddf.catalog.ui.query.monitor.impl;
+
+import static org.apache.commons.lang3.Validate.notBlank;
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.codice.ddf.catalog.ui.query.monitor.api.SubscriptionsPersistentStore;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Associate sets of emails with an ID. This implementation will preserve any other
+ * properties stored under the ID.
+ */
+public class SubscriptionsPersistentStoreImpl extends AbstractSubscriptionsPersistentStore {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(SubscriptionsPersistentStore.class);
+
+    /**
+     * The persistent store type.
+     */
+    private static final String TYPE = PersistentStore.SUBSCRIPTIONS_TYPE;
+
+    private static final String EMAIL_PROPERTY = "email";
+
+    private static final String ID = "id";
+
+    private final PersistentStore persistentStore;
+
+    /**
+     * @param persistentStore must be non-null
+     */
+    public SubscriptionsPersistentStoreImpl(PersistentStore persistentStore) {
+        notNull(persistentStore, "persistentStore must be non-null");
+        this.persistentStore = persistentStore;
+    }
+
+    private List<Map<String, Object>> get(String id) throws PersistenceException {
+        List<Map<String, Object>> results = persistentStore.get(TYPE, query(id));
+        assert results.size() <= 1;
+        return results;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void addEmails(String id, Set<String> emails) {
+        notBlank(id, "id must be non-blank");
+        notNull(emails, "emails must be non-null");
+        emails.forEach(email -> notBlank(email, "emails in set must be non-blank"));
+
+        try {
+            List<Map<String, Object>> results = get(id);
+
+            if (!results.isEmpty()) {
+                PersistentItem item = convert(results.get(0));
+
+                if (item.containsKey(EMAIL_PROPERTY + PersistentItem.TEXT_SUFFIX)) {
+                    Set<String> newValue = new HashSet<>(emails);
+                    Object value = item.get(EMAIL_PROPERTY + PersistentItem.TEXT_SUFFIX);
+                    if (value instanceof String) {
```

Comment:
Aren't you only storing `Set`s here (i.e., is it possible for this to be a string)?